### PR TITLE
Admin Flow

### PR DESCRIPTION
**Pull request by SnehdeepMohite**

### CODE REVIEW

File: app/assets/javascripts/angular_apps/admin/controllers/panel.controller.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        var credentials = $scope.invite;
+
+        var config = {
+            headers: {
+                'X-HTTP-Method-Override': 'POST'
+            }
+        };
+
+        Auth.register(credentials, config).then(function(credentials) {
+            $window.location.href = ('/admin#/index');
```
Comment (umar-siddiqui) [inline app/assets/javascripts/angular_apps/admin/controllers/panel.controller.js:37]: Use State.go

File: app/assets/javascripts/angular_apps/admin/controllers/panel.controller.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        }, function(error) {
+            alert('Something went wrong');
+        });
+      };
+
+      $scope.inviteUser = function(){
+        $window.location.href = ('/admin#/invite');
```
Comment (umar-siddiqui) [inline app/assets/javascripts/angular_apps/admin/controllers/panel.controller.js:44]: Use State.go

File: app/assets/javascripts/angular_apps/admin/controllers/panel.controller.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      };
+
+      $scope.assesmentReport = function(){
+        $window.location.href = ('/admin#/report');
```
Comment (umar-siddiqui) [inline app/assets/javascripts/angular_apps/admin/controllers/panel.controller.js:48]: Use State.go

File: app/assets/javascripts/angular_apps/dashboard/controllers/dashboard.controller.js [hunk 4]
```diff
@@ -4,11 +4,11 @@
   angular
     .module('ungc.dashboard')
     .controller('DashboardController',[
-      '$scope', 'DashboardService',
+      '$scope', 'Auth', 'DashboardService',
```
Comment (umar-siddiqui) [inline app/assets/javascripts/angular_apps/dashboard/controllers/dashboard.controller.js:7]: Why is auth required in this controller please remove it

File: app/assets/javascripts/angular_apps/dashboard/controllers/dashboard.controller.js [hunk 5]
```diff
@@ -23,4 +23,5 @@
       };
     }
 
+
```
Comment (umar-siddiqui) [inline app/assets/javascripts/angular_apps/dashboard/controllers/dashboard.controller.js:26]: Y added this space ??

File: app/mailers/application_mailer.rb [hunk 6]
```diff
@@ -0,0 +1,4 @@
+class ApplicationMailer < ActionMailer::Base
+  default from: "sunny@treeni.com"
```
Comment (umar-siddiqui) [inline app/mailers/application_mailer.rb:2]: replace the email with a more relevant sender. selfassesment@unglobalcompact.org or something.

File: app/mailers/notifier.rb [hunk 7]
```diff
@@ -0,0 +1,9 @@
+class Notifier < ApplicationMailer
+  default from: 'sunny@treeni.com',
+          return_path: 'sunny@treeni.com'
```
Comment (umar-siddiqui) [inline app/mailers/notifier.rb:3]: replace the email with a more relevant sender. selfassessment@unglobalcompact.org or something.

File: app/controllers/assesments_controller.rb [hunk 8]
```diff
@@ -7,4 +7,14 @@ def index
     render json: assesments,
            each_serializer: AssesmentsSerializers::IndexSerializer
   end
+
+  def all_assesments
+    respond_to do |format|
+      format.html
```
Comment (umar-siddiqui) [inline app/controllers/assesments_controller.rb:13]: is this really needed out here

File: app/serializers/admin_serializers/admin_index_serializer.rb [hunk 9]
```diff
@@ -0,0 +1,14 @@
+module AdminSerializers
+  # class AdminIndexSerializer
+  class AdminIndexSerializer < ActiveModel::Serializer
```
Comment (umar-siddiqui) [inline app/serializers/admin_serializers/admin_index_serializer.rb:3]: Please add a relevant name as discussed for naming serializers in this project. Module name should be the controller name and serailzer name should be the action name

File: app/serializers/admin_serializers/users_listing_serializer.rb [hunk 10]
```diff
@@ -0,0 +1,31 @@
+module AdminSerializers
+  # class AdminIndexSerializer
+  class UsersListingSerializer < ActiveModel::Serializer
```
Comment (umar-siddiqui) [inline app/serializers/admin_serializers/users_listing_serializer.rb:3]: Please add a relevant name as discussed for naming serializers in this project. Module name should be the controller name and serailzer name should be the action name

File: app/serializers/dashboard_serializer.rb [hunk 11]
```diff
@@ -1,4 +1,4 @@
 # class DashboardSerializer
 class DashboardSerializer < ActiveModel::Serializer
-  attributes :id, :created_at, :name , :submitted_at , :user_ids
+  attributes :id, :created_at, :name, :description, :submitted_at, :user_ids
```
Comment (umar-siddiqui) [inline app/serializers/dashboard_serializer.rb:3]: Please move serializer to its proper semantic position

File: config/routes.rb [hunk 12]
```diff
@@ -1,5 +1,5 @@
 Rails.application.routes.draw do
-  devise_for :users
+  devise_for :users, controllers: {registrations: 'registrations'}
```
Comment (umar-siddiqui) [inline config/routes.rb:2]: Rubo cop error on this line

File: config/routes.rb [hunk 13]
```diff
@@ -12,13 +12,25 @@
...
+  end
+
   resources :dashboard, only: [:index]
-  resources :assesments, only: [:index, :create]
+
+  resources :users, only: [:index]
+
+  resources :assesments, only: [:index, :create] do
+    collection do
+      get :all_assesments
```
Comment (umar-siddiqui) [inline config/routes.rb:26]: this action has to reside in a different name space that only belong to an admin. i.e is there will be a different set of admin controllers here

File: app/assets/javascripts/angular_apps/authentication/controllers/sign_in.controller.js [hunk 14]
```diff
@@ -17,8 +17,14 @@
...
-        $window.location.href = '/dashboard#/';
-        alert('Successfully signed in user!')
+        if(user.type == 'admin'){
+          console.log('Hi Admin');
+          $window.location.href = 'admin#/';
+          alert('ADMIN Successfully signed in!')
+        } else{
+          $window.location.href = '/dashboard#/';
+          alert('Successfully signed in user!')
+        };
```
Comment (umar-siddiqui) [inline app/assets/javascripts/angular_apps/authentication/controllers/sign_in.controller.js:27]: after login in u did not handle a single case for the admin or the normal user to be authenticated on there respective pages. i.e Current admin can access all normal user pages and normal user can access admin pages.

File: app/assets/javascripts/angular_apps/common/promise_factory.js [hunk 15]
```diff
@@ -2,7 +2,7 @@
   'use strict';
 
   angular
-    .module('ungc.dashboard')
+    .module('ungc.admin')
```
Comment (umar-siddiqui) [inline app/assets/javascripts/angular_apps/common/promise_factory.js:5]: why does common module only belong to admin module. What is the significance of common out here.

File: app/assets/javascripts/angular_apps/common/utility_factory.js [hunk 16]
```diff
@@ -2,7 +2,7 @@
   'use strict';
 
   angular
-    .module('ungc.dashboard')
+    .module('ungc.admin')
```
Comment (umar-siddiqui) [inline app/assets/javascripts/angular_apps/common/utility_factory.js:5]: why does common module only belong to admin module. What is the significance of common out here.

File: config/environments/development.rb [hunk 17]
```diff
@@ -16,6 +16,19 @@
...
+  # config.action_mailer.delivery_method = :smtp
+  # config.action_mailer.smtp_settings = {
+  #   :address              => "smtp.gmail.com",
+  #   :port                 => 587,
+  #   :user_name            => 'email@weboapps.com',
+  #   :password             => 'weboqa6186',
+  #   :authentication       => "plain",
+  #   :enable_starttls_auto => true
+  # }
+
```
Comment (umar-siddiqui) [inline config/environments/development.rb:31]: kindly remove the comment code

File: app/serializers/admin_serializers/users_listing_serializer.rb [hunk 18]
```diff
@@ -0,0 +1,31 @@
...
+    attributes :_id, :name, :email
+
+    has_many :assesments
+    has_one :company
+
+    def company
+      object.company
+    end
```
Comment (umar-siddiqui) [inline app/serializers/admin_serializers/users_listing_serializer.rb:11]: this method is not required.  by default serializer does this

File: app/serializers/admin_serializers/users_listing_serializer.rb [hunk 19]
```diff
@@ -0,0 +1,31 @@
...
+
+    def assesments
+      object.assesments.map do |a|
+        {
+          _id: a._id.to_s,
+          id: a.id.to_s,
```
Comment (umar-siddiqui) [inline app/serializers/admin_serializers/users_listing_serializer.rb:17]: why do you need .id when there is already _id. and to_s is not required.

File: app/serializers/admin_serializers/users_listing_serializer.rb [hunk 20]
```diff
@@ -0,0 +1,31 @@
...
+
+    has_many :assesments
+    has_one :company
+
+    def company
+      object.company
+    end
+
+    def assesments
+      object.assesments.map do |a|
```
Comment (umar-siddiqui) [inline app/serializers/admin_serializers/users_listing_serializer.rb:14]: what does 'a' mean out here.please u more relevant variable name here.

File: app/controllers/assesments_controller.rb [hunk 21]
```diff
@@ -7,4 +7,14 @@ def index
     render json: assesments,
            each_serializer: AssesmentsSerializers::IndexSerializer
   end
+
+  def all_assesments
```
Comment (umar-siddiqui) [inline app/controllers/assesments_controller.rb:11]: this action shld exists in admin name space in a separate controller

File: app/controllers/users_controller.rb [hunk 22]
```diff
@@ -0,0 +1,8 @@
+# class UsersController
+class UsersController < ApplicationController
+  def index
+    users = User.all
+    render json: users,
+           each_serializer: AdminSerializers::UsersListingSerializer
```
Comment (umar-siddiqui) [inline app/controllers/users_controller.rb:6]: please use a more relevant name for serializer

File: app/controllers/admin/panel_controller.rb [hunk 23]
```diff
@@ -0,0 +1,14 @@
+module Admin
+# class PanelController
```
Comment (umar-siddiqui) [inline app/controllers/admin/panel_controller.rb:2]: please align comment.... rubo cop error

File: app/assets/javascripts/angular_apps/admin/controllers/panel.controller.js (hunk 1 above)
Comment (SnehdeepMohite) [inline app/assets/javascripts/angular_apps/admin/controllers/panel.controller.js:37]: implemented

File: app/assets/javascripts/angular_apps/admin/controllers/panel.controller.js (hunk 2 above)
Comment (SnehdeepMohite) [inline app/assets/javascripts/angular_apps/admin/controllers/panel.controller.js:44]: implemented

File: app/assets/javascripts/angular_apps/admin/controllers/panel.controller.js (hunk 3 above)
Comment (SnehdeepMohite) [inline app/assets/javascripts/angular_apps/admin/controllers/panel.controller.js:48]: implemented

File: app/assets/javascripts/angular_apps/dashboard/controllers/dashboard.controller.js (hunk 4 above)
Comment (SnehdeepMohite) [inline app/assets/javascripts/angular_apps/dashboard/controllers/dashboard.controller.js:7]: removed

File: app/assets/javascripts/angular_apps/dashboard/controllers/dashboard.controller.js (hunk 5 above)
Comment (SnehdeepMohite) [inline app/assets/javascripts/angular_apps/dashboard/controllers/dashboard.controller.js:26]: Space removed

File: app/controllers/admin/panel_controller.rb (hunk 23 above)
Comment (SnehdeepMohite) [inline app/controllers/admin/panel_controller.rb:2]: implemented

File: app/mailers/application_mailer.rb (hunk 6 above)
Comment (SnehdeepMohite) [inline app/mailers/application_mailer.rb:2]: resolved

File: app/mailers/notifier.rb (hunk 7 above)
Comment (SnehdeepMohite) [inline app/mailers/notifier.rb:3]: resolved

File: app/serializers/admin_serializers/users_listing_serializer.rb (hunk 20 above)
Comment (SnehdeepMohite) [inline app/serializers/admin_serializers/users_listing_serializer.rb:14]: 'a' replaced with 'assesment'

File: app/serializers/admin_serializers/users_listing_serializer.rb (hunk 19 above)
Comment (SnehdeepMohite) [inline app/serializers/admin_serializers/users_listing_serializer.rb:17]: ok. Implemented

File: config/environments/development.rb (hunk 17 above)
Comment (SnehdeepMohite) [inline config/environments/development.rb:31]: removed

File: config/routes.rb (hunk 12 above)
Comment (SnehdeepMohite) [inline config/routes.rb:2]: resolved

File: app/controllers/assesments_controller.rb (hunk 8 above)
Comment (SnehdeepMohite) [inline app/controllers/assesments_controller.rb:13]: No. Removed

File: app/controllers/users_controller.rb (hunk 22 above)
Comment (SnehdeepMohite) [inline app/controllers/users_controller.rb:6]: implemented

File: app/serializers/admin_serializers/users_listing_serializer.rb (hunk 18 above)
Comment (SnehdeepMohite) [inline app/serializers/admin_serializers/users_listing_serializer.rb:11]: ok. Removed and checked it's working

File: app/serializers/admin_serializers/users_listing_serializer.rb (hunk 10 above)
Comment (SnehdeepMohite) [inline app/serializers/admin_serializers/users_listing_serializer.rb:3]: serializer's name changed to AllUsersSerializer

File: config/routes.rb (hunk 13 above)
Comment (SnehdeepMohite) [inline config/routes.rb:26]: [12:11:37 PM] Umar Siddiqui: ab ke liye ye chhod

:)

**Pull request merged**